src/components/calcite-input/calcite-input.tsx

```diff
@@ -652,6 +724,12 @@ export class CalciteInput implements LabelableComponent, FormComponent {
     window.clearInterval(this.nudgeNumberValueIntervalId);
   };
 
+  private warnAboutInvalidNumberValue(value: string): void {
+    if (this.type === "number" && value && !isValidNumber(value)) {
+      console.warn(`The specified value "${value}" cannot be parsed, or is out of range.`);
+    }
+  }
```

Comment:
This mimics what the native number input does when you try to set the value to an invalid number.